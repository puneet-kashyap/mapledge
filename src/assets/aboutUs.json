{
  "benefits": [
    {
      "title": "Cost-effective temporary staffing",
      "desc": "Leverage our temporary professionals during peak workload periods, which reduces overhead costs and increases efficiency."
    },
    {
      "title": "Fast hiring",
      "desc": "We have highly skilled, pre-evaluated job candidates ready to begin working right away"
    },
    {
      "title": "Flexible staffing solutions",
      "desc": "Whether you need one person or an entire team, we have professionals with a wide array of expertise."
    },
    {
      "title": "Customer-first approach",
      "desc": "We are dedicated to providing superior customer service every step of the way. That’s why 9 out of 10 clients tell us they’d recommend us to their colleagues."
    },
    {
      "title": "Competitive pay",
      "desc": "We know what fair and competitive compensation is for your area and work with our clients to help keep them educated on the market value for in-demand skills."
    }
  ]
}