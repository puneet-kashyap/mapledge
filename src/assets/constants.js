const assets = {
  companyName: 'Mapledge Staffing',
  contacts : [
    {
      name: 'Neeraj Tara',
      phone: '+1-306-999-5295',
      email: 'neerajtara18@gmail.com'
    },
    {
      name: 'Varun Tara',
      phone: '+1-306-999-5295',
      email: 'taravarun@gmail.com'
    }
  ]
}

export default assets;