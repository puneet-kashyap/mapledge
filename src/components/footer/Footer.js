import React from 'react';
import './footer.css';
import assets from '../../assets/constants';

class Footer extends React.Component {
  contacts = [];

  render() {
    return (
      <>
        <footer className="footer">
          {assets.contacts.map((item, index) => {
            return (
              <div key={index} className="footer-item">
                <div className="name">{item.name}</div>
                <div className="contact-option">
                  <a className="phone" href={`tel:${item.phone}`}>
                    {item.phone}
                  </a>
                </div>
                <div className="contact-option">
                  <a className="email" href={`mailto:${item.email}`}>
                    {item.email}
                  </a>
                </div>
              </div>
            );
          })}
        </footer>
        {subFooter(assets.contacts[0].email)}
      </>
    );
  }
}

const subFooter = email => {
  return (
    <div className="footer-below">
      <div>
        Copyright &copy; {assets.companyName} {new Date().getFullYear()}
      </div>
      <div>
        Built with{' '}
        <span role="img" aria-label="heart">
          ðŸ’–
        </span>
        by the
        <a style={{ color: 'white' }} href={`mailto:${email}`}>
          Kashyap
        </a>
        team
      </div>
    </div>
  );
};

export default Footer;
