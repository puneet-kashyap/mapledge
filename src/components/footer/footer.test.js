import React from 'react';
import { shallow, configure } from 'enzyme';
import Footer from './Footer';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import assets from '../../assets/constants';

configure({ adapter: new Adapter() });
const wrapper = shallow(<Footer />);

it('Footer component renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Footer />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should display all the contacts info', () => {
  wrapper.find('.name').forEach((node, index) => {
    expect(node.text()).toBe(assets.contacts[index].name);
  });
  wrapper.find('.phone').forEach((node, index) => {
    expect(node.text()).toBe(assets.contacts[index].phone);
  });
  wrapper.find('.email').forEach((node, index) => {
    expect(node.text()).toBe(assets.contacts[index].email);
  });
});

it('should display sub footer', () => {
  expect(wrapper.find('.footer-below').text()).toContain(assets.companyName)
})
