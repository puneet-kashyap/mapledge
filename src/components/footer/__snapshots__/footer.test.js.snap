// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
Array [
  <footer
    className="footer"
  >
    <div
      className="footer-item"
    >
      <div
        className="name"
      >
        Inquiries
      </div>
      <div
        className="contact-option"
      >
        <a
          className="phone"
          href="tel:+1-306-999-5295"
        >
          +1-306-999-5295
        </a>
      </div>
      <div
        className="contact-option"
      >
        <a
          className="email"
          href="mailto:info@mapledge.ca"
        >
          info@mapledge.ca
        </a>
      </div>
    </div>
    <div
      className="footer-item"
    >
      <div
        className="name"
      >
        Human Resource
      </div>
      <div
        className="contact-option"
      >
        <a
          className="phone"
          href="tel:+1-306-999-5295"
        >
          +1-306-999-5295
        </a>
      </div>
      <div
        className="contact-option"
      >
        <a
          className="email"
          href="mailto:hr@mapledge.ca"
        >
          hr@mapledge.ca
        </a>
      </div>
    </div>
  </footer>,
  <div
    className="footer-below"
  >
    <div>
      Copyright Â© 
      Mapledge Staffing
       
      2019
    </div>
    <div>
      Built with
       
      <span
        aria-label="heart"
        role="img"
      >
        ðŸ’–
      </span>
      by the
      <a
        href="mailto:info@mapledge.ca"
        style={
          Object {
            "color": "white",
          }
        }
      >
        Kashyap
      </a>
      team
    </div>
  </div>,
]
`;
